### PR TITLE
Add Identity With Delegations to Common Types

### PR DESCRIPTION
**Pull request by shavital7**
Adding a new type for Identity with delegations, needed for backing resources, based on the [HOBO v2 documentation](https://armwiki.azurewebsites.net/authorization/managed_identities/hosted_on_behalf_of_v2/BackingResourceProviderOnboarding.html)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify
      Adding a new swagger type to common
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
      No deployment needed
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
      July2022
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by aviaoh** [comment]
@leni-msft can we please merge it? should we wait for anything else other than ARM signoff?

**Review by leniatgh — APPROVED** [review]

File: specification/common-types/resource-management/v4/managedidentitywithdelegation.json
Comment (leni-msft) [inline specification/common-types/resource-management/v4/managedidentitywithdelegation.json:10]: These properties are only internally available to other RPs, but not to end users. @msyyc @ArcturusZhang @qiaozha Is it possible to avoid exposing these properties in SDKs? Is the description enough?

Comment (aviaoh) [reply specification/common-types/resource-management/v4/managedidentitywithdelegation.json:10]: and also hide it in documentation.
i see it's hidden in https://docs.microsoft.com/en-us/rest/api/sql/2021-11-01-preview/databases/create-or-update?tabs=HTTP#databaseidentity somehow.

Comment (leni-msft) [reply specification/common-types/resource-management/v4/managedidentitywithdelegation.json:10]: They're not defining `delegatedResources` in their swagger:
https://github.com/Azure/azure-rest-api-specs/blob/8c2258114e565bb041dcb25a761acb965a4870e7/specification/sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/Databases.json#L821-L851

Comment (aviaoh) [reply specification/common-types/resource-management/v4/managedidentitywithdelegation.json:10]: sorry, i was looking on master rather than main.

so question remains the same - Is it possible to avoid exposing these properties in SDKs and documentation?

**Comment by aviaoh** [comment]
> Hi, @shavital7. Your PR has no update for 14 days and it is marked as stale PR. If no further update for over 14 days, the bot will close the PR. If you want to refresh the PR, please remove `no-recent-activity` label.

we are waiting for @leni-msft and @raych1 response on this PR.
anything new here?

**Comment by leni-msft** [comment]
@aviaoh Per discussion with @raych1 and @sanjaiganesh , it's ok to merge it as ARM call out it will be used for all RPs internally

**Pull request merged**
